src/util/locking/FileSystemResourceLocker.ts

```diff
@@ -0,0 +1,160 @@
+  lockDirectory?: string;
+  /** Custom settings concerning retrying locks */
+  attemptSettings?: AttemptSettings;
+}
+
+function isCodedError(err: unknown): err is { code: string } & Error {
+  return typeof err === 'object' && err !== null && 'code' in err;
+}
+
+/**
+ * A resource locker making use of the [proper-lockfile](https://www.npmjs.com/package/proper-lockfile) library.
+ * Note that no locks are kept in memory, thus this is considered thread- and process-safe.
+ *
+ * This **proper-lockfile** library has its own retry mechanism for the operations, since a lock/unlock call will
+ * either resolve successfully or reject immediately with the causing error. The retry function of the library
+ * however will be ignored and replaced by our own LockUtils' {@link retryFunctionUntil} function.
+ *
+ * This allows us to reuse the AttemptSettings semantics similar to the {@link RedisLocker} class.
+ */
+export class FileSystemResourceLocker implements ResourceLocker, Finalizable {
+  protected readonly logger = getLoggerFor(this);
+  private readonly attemptSettings: Required<AttemptSettings>;
+  /** Folder that stores the locks */
+  private readonly lockFolder: string;
+
+  /**
+   * Create a new FileSystemResourceLocker
+   * @param rootFilePath - The rootPath of the filesystem _[default is the current dir `./`]_
+   * @param lockDirectory - The path to the directory where locks will be stored (appended to rootFilePath)
+                            _[default is `/.internal/locks`]_
+   * @param attemptSettings - Custom settings concerning retrying locks
+   */
+  public constructor(args: FileSystemResourceLockerArgs = {}) {
+    const { rootFilePath, lockDirectory, attemptSettings } = args;
+    this.attemptSettings = { ...attemptDefaults, ...attemptSettings };
+    this.lockFolder = joinFilePath(rootFilePath ?? './', lockDirectory ?? '/.internal/locks');
+    ensureDirSync(this.lockFolder);
+  }
+
+  /**
+   * Wrapper function for all un/lock operations. Any errors coming from the `fn()` will be swallowed.
+   * Only `ENOTACQUIRED` errors wills be thrown (trying to release lock that didn't exist).
+   * @param fn - The function reference to swallow errors from.
+   */
+  private swallowErrors(fn: () => Promise<VoidFunctionOrVoid>): () => Promise<BooleanOrVoid> {
+    return async(): Promise<BooleanOrVoid> => {
+      try {
+        await fn();
+        return true;
+      } catch (err: unknown) {
+        // Only this error should be thrown
+        if (isCodedError(err) && err.code === 'ENOTACQUIRED') {
+          throw err;
+        }
+      }
+    };
+  }
+
+  public async acquire(identifier: ResourceIdentifier): Promise<void> {
+    const { path } = identifier;
+    this.logger.debug(`Acquiring lock for ${path}`);
+    try {
+      const opt = this.generateOptions(identifier, defaultLockOptions);
+      await retryFunction(
+        this.swallowErrors((): Promise<VoidFunctionOrVoid> => lock(path, opt)),
```

Comment:
```suggestion
        this.swallowErrors(lock.bind(null, path, opt)),
```
This should also work. Not that it really matters in practice, it's mostly me being enthusiastic about being able to use `bind` since there are not that many relevant cases. And I think the result is one less call.